diff --git a/expo-module-gradle-plugin/src/main/kotlin/expo/modules/plugin/gradle/ExpoGradleHelperExtension.kt b/expo-module-gradle-plugin/src/main/kotlin/expo/modules/plugin/gradle/ExpoGradleHelperExtension.kt
index 7c387abb5aeaf058406947347e4aeab4cf076ee4..8538d553ddffa66e08842fcf54c7f57fdb2c9a0a 100644
--- a/expo-module-gradle-plugin/src/main/kotlin/expo/modules/plugin/gradle/ExpoGradleHelperExtension.kt
+++ b/expo-module-gradle-plugin/src/main/kotlin/expo/modules/plugin/gradle/ExpoGradleHelperExtension.kt
@@ -43,7 +43,7 @@ open class ExpoGradleHelperExtension {
     reactNativeDir = reactNativeDirFromSource ?: File(
       project.providers.exec { env ->
         env.workingDir(project.rootDir)
-        env.commandLine("node", "--print", "require.resolve('react-native/package.json')")
+        env.commandLine(System.getenv("BUN_EXECUTABLE") ?: "bun", "--print", "require.resolve('react-native/package.json')")
       }.standardOutput.asText.get().trim()
     ).parentFile
 
