diff --git a/android/build.gradle.kts b/android/build.gradle.kts
index 2e01292a695638f4b82d27531d82c0670c149075..16f99160cd49ca0bc48e9e0da7a28d6d2b83c53b 100644
--- a/android/build.gradle.kts
+++ b/android/build.gradle.kts
@@ -11,6 +11,8 @@ plugins {
     id("org.jetbrains.kotlin.android")
 }
 
+val jsRuntimeExecutable = System.getenv("BUN_EXECUTABLE") ?: "bun"
+
 fun safeExtGet(prop: String, fallback: Any?): Any? =
     if (rootProject.extensions.extraProperties.has(prop)) rootProject.extensions.extraProperties.get(prop) else fallback
 
@@ -32,7 +34,7 @@ fun resolveReactNativeDirectory(): File {
     val reactNativePackage = file(
         providers.exec {
             workingDir(rootDir)
-            commandLine("node", "--print", "require.resolve('react-native/package.json')")
+            commandLine(jsRuntimeExecutable, "--print", "require.resolve('react-native/package.json')")
         }.standardOutput.asText.get().trim()
     )
     if (reactNativePackage.exists()) {
@@ -298,7 +300,7 @@ androidComponents {
 
 val validateReactNativeVersionResult = providers.exec {
     workingDir(projectDir.path)
-    commandLine("node", "./../scripts/validate-react-native-version.js", REACT_NATIVE_VERSION)
+    commandLine(jsRuntimeExecutable, "./../scripts/validate-react-native-version.js", REACT_NATIVE_VERSION)
     isIgnoreExitValue = true
 }
 
